opba-db/src/test/java/de/adorsys/opba/db/BankProtocolActionsSqlGeneratorTest.java

```diff
@@ -0,0 +1,149 @@
+package de.adorsys.opba.db;
+
+import com.google.common.io.FileWriteMode;
+import com.google.common.io.Files;
+import com.google.common.io.Resources;
+import de.adorsys.opba.db.config.TestConfig;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.util.List;
+import java.util.UUID;
+
+import static de.adorsys.opba.db.BankProtocolActionsSqlGeneratorTest.ENABLE_BANK_PROTOCOL_ACTIONS_SQL_GENERATION;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Is not actually a test. This class generates 'banks_random_data.csv' out of 'banks.csv',
+ * which contains sql insert statements into 'opb_bank_action' and 'opb_bank_sub_action' tables
+ */
+@SpringBootTest(classes = TestConfig.class)
+@EnabledIfEnvironmentVariable(named = ENABLE_BANK_PROTOCOL_ACTIONS_SQL_GENERATION, matches = "true")
+public class BankProtocolActionsSqlGeneratorTest {
+    public static final String ENABLE_BANK_PROTOCOL_ACTIONS_SQL_GENERATION = "ENABLE_BANK_PROTOCOL_ACTIONS_SQL_GENERATION";
+
+    private static final String BANK_DATA_SOURCE_PATH = "migration/migrations/banks.csv";
+
+    private static final String BANK_ACTION_DESTINATION_PATH = "src/main/resources/migration/migrations/bank_action_data.csv";
+    private static final String BANK_SUB_ACTION_DESTINATION_PATH = "src/main/resources/migration/migrations/bank_sub_action_data.csv";
+    private static final String BANK_PROFILE_DESTINATION_PATH = "src/main/resources/migration/migrations/bank_profile_data.csv";
+
+    private static final String BANK_ACTION_CSV_HEADER = "id,bank_uuid,protocol_action,protocol_bean_name,consent_supported";
+    private static final String BANK_SUB_ACTION_CSV_HEADER = "id,action_id,protocol_action,sub_protocol_bean_name";
+    private static final String BANK_PROFILE_CSV_HEADER = "uuid,name,bic,url,adapter_id,bank_code,idp_url,aspsp_sca_approaches";
+
+    @Value("${bank-action-generator.action.start-id}")
+    private Integer bankActionId;
+
+    @Value("${bank-action-generator.sub-action.start-id}")
+    private Integer bankSubActionId;
+
+    @Test
+    @SneakyThrows
+    public void convertToDbSql() {
+        List<String> banks = removeXs2aBanks(readResourceLines(BANK_DATA_SOURCE_PATH));
+        prepareDestinationFiles();
+
+        for (String bank : banks) {
+            writeXs2aBankActionData(bank);
+            writeHbciBankData(bank);
+        }
+    }
+
+    private List<String> removeXs2aBanks(List<String> banks) {
+        return banks.subList(7, banks.size());
```

Comment:
Introduce property instead of hardcoded `7` value

Reply:
'bank-action-generator.skip-banks' property was added